lib/streamlit/runtime/metrics_util.py

```diff
@@ -206,16 +198,73 @@ def _get_command_telemetry(callable: Callable[..., Any], *args, **kwargs) -> Com
     return Command(name=name, args=arguments)
 
 
-def to_microseconds(seconds):
+def to_microseconds(seconds: float) -> int:
+    """Convert seconds into microseconds."""
     return int(seconds * 1000000)
 
 
 F = TypeVar("F", bound=Callable[..., Any])
 
 
-def gather_metrics(callable: F) -> F:
-    @wraps(callable)
-    def wrap(*args, **kwargs):
+@overload
+def gather_metrics(
+    func: F,
+    name: Optional[str] = None,
+) -> F:
```

Comment:
Good point. I will think about this tomorrow. I'm not 100% convinced yet because the metrics tracking should actually reflect the exact namings of our public API and if something gets renamed, it should change the name as well. The tests are making sure that we are tracking everything from our public API (with the correct names). 

It's mostly for the few internal methods that `gather_metrics` with `name` is recommended to not run into the issue stated by you.

Reply:
Decided to enforce the `name` 👍